src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -1862,3 +1862,88 @@ end
 --wrapper for createButton 
 -- createButton is deprecated better use createLabel instead
 createButton = createLabel
+
+
+-- Internal function used by copy2html and copy2decho
+local function copy2color(name,win,str,inst)
+  local line = getCurrentLine(win or "main")
+  if (not str and line == "ERROR: mini console does not exist") or type(str) == "number" then
+    win, str, inst = "main", win, str
+    line = getCurrentLine(win)
+  end
+  win = win or "main"
+  str = str or line
+  inst = inst or 1
+  local start, len = selectString(win, str, inst), #str
+  if not start then
+    error(name..": string not found",3)
+  end
+  local style, endspan, result, r, g, b, br, bg, bb, cr, cg, cb, crb, cgb, cbb
+  if name == "copy2html" then
+    style = "%s<span style=\'color: rgb(%d,%d,%d);background: rgb(%d,%d,%d);'>%s"
+    endspan = "</span>"
+  elseif name == "copy2decho" then
+    style = "%s<%d,%d,%d:%d,%d,%d>%s"
+    endspan = "<r>"
+  end
+  for index = start + 1, start + len do
+    if win ~= "main" then
+      selectSection(win, index - 1, 1)
+      r,g,b = getFgColor(win)
+      rb,gb,bb = getBgColor(win)
+    else
+      selectSection(index - 1, 1)
+      r,g,b = getFgColor()
+      rb,gb,bb = getBgColor()
+    end
+    
+    if not table.is_empty(table.complement({r,g,b,rb,gb,bb},{cr,cg,cb,crb,cgb,cbb})) then
+      cr,cg,cb,crb,cgb,cbb = r,g,b,rb,gb,bb
+      result = string.format(style, result and (result..endspan) or "", r, g, b, rb, gb, bb, line:sub(index, index))
+    else
+      result = result .. line:sub(index, index)
+    end
+  end
+  result = result .. endspan
+  if name == "copy2html" then
+    local conversions = {["¦"] = "&brvbar;", ["×"] = "&times;", ["«"] = "&#171;", ["»"] = "&raquo;"}
```

Comment:
I was going to suggest this and then realised that's the code from my screendump package :grin: